### PR TITLE
Viewer library

### LINKED ISSUE #11: displaying a Google Slides doc in a carousel
Georgia (et al) would like to be able to 
(a) display a google slides doc without any hassles, and
(b) to include one in a carousel

### LINKED ISSUE #53: Integrate the carousel library into Viewer.
- [x] Create modules for [Viewer](https://github.com/ftlabs/screens-viewer) and [Carousel](https://github.com/ftlabs/screens-carousel) Which can be shared with the extension
- [x] Write tests and setup automated testing for those modules
- [x] Replace the existing implementation with the viewer module #52 
- [x] Integrate the Carousel Library so that it is used if a carousel is detected. [Example.](http://git.svc.ft.com/projects/LABSPROT/repos/screens-extension/browse/app/_scripts/background.js#208)

### PR DESCRIPTION
**Pull request by AdaRoseCannon**
This is for using the viewer module. Which interfaces with the api and handles scheduling.

~~Don't merge yet there are no tests written for the viewer library.~~

~~This keeps the same functionality (no carousel integration) but uses the new library.~~

Tests are added to viewer now
Added carousel too.

Need to fix screens tests now...

Fixes #11
Fixes #53

### CODE REVIEW

**Comment by seanmtracey** [comment]
Looks good to me.

**Comment by AdaRoseCannon** [comment]
It says tests failing but, `npm run test` passes all tests.

File: client/viewer/js/main.js [hunk 1]
```diff
@@ -1,75 +1,95 @@
...
-});
-
-socket.on('disconnect', function() {
-	socket.io.reconnect();
-	viewer.setConnectionState(false);
-});
+const Viewer = require('ftlabs-screens-viewer');
+const Carousel = require('ftlabs-screens-carousel');
+const port = location.port ? ':'+location.port : '';
+const host = '//'+location.hostname+port;
```
Comment (JakeChampion) [inline client/viewer/js/main.js]: I think `location.origin` would be a better fit.
`location.origin` returns the protocol, hostname and port --
"http://local.ft.com:3000"

Comment (AdaRoseCannon) [inline client/viewer/js/main.js]: :+1:

File: client/viewer/js/main.js [hunk 2]
```diff
@@ -1,75 +1,95 @@
...
-});
+function viewerIsRunningInElectron() {
+	return (navigator.userAgent.indexOf('Electron') > 0 && navigator.userAgent.indexOf('FTLabs-Screens') > 0);
+}
 
-socket.on('reload', function(){
-	window.location.reload();
-});
+// Called by the script loader once the page has loaded
+window.screensInit = function() {
```
Comment (JakeChampion) [inline client/viewer/js/main.js:13]: What do you think about exporting this function as the modules interface?

Comment (AdaRoseCannon) [inline client/viewer/js/main.js:13]: We'd need to do it anyway as it is part of the original script loading code.

Also the viewer does not handle any DOM it is a general interface for the websockets api.

**Pull request merged**